generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -92,6 +92,9 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
     if (shadingLanguageVersion.supportedMixNonfloatBool()) {
       expressionIdentities.add(new IdentityMixBvec());
     }
+    if (shadingLanguageVersion.supportedNonSquareMatrices()) {
```

Comment:
Seems limiting to only do this if non-square matrices are supported, since e.g. we can put a mat2x2 into a mat3x3.

It would be better to have the code that applies the identity decide whether to allow square matrices or not.

Reply:
The suggested changes to the conditional logic make this much easier to handle inside of the function, so I've implemented this.